test/script/run_tests.sh

```diff
@@ -47,12 +47,10 @@ run_test_suite () {
 
   npm run test:deps
 
-  if [ -z "$COVERAGE" ]
-  then
-    node test/test.js
-  else
-    nyc node test/test.js
-  fi
+  rm -rf ./test_output
+  mkdir ./test_output
+  nyc node test/test.js -o ./test_output
+  ls test_output/*.tap | while read f; do cat $f | ./node_modules/.bin/tap-junit > $f.junit.xml; done
```

Comment:
Changes to this script will change the behaviour of `npm test`, which is used for local development. I don't think we want to have (a) coverage and (b) generation of .tap -> .junit.xml files in the general case. 

1. Currently the "Test" action does *not* run with coverage (i.e. does not run with `nyc ...`). Is there a particular reason to add coverage testing for these runs?
2. How about we put the `-o ./test_output` usage and conversion to junit format handling behind a guard on the "CI" envvar being defined? This would mean that `npm test` for local dev would not be changed, but when run in GitHub Actions (which [defines CI=true always](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables)) the Junit files would be defined for the OTel export.

Something like this (untested):

```diff
--- a/test/script/run_tests.sh
+++ b/test/script/run_tests.sh
@@ -47,11 +47,21 @@ run_test_suite () {

   npm run test:deps

-  if [ -z "$COVERAGE" ]
-  then
-    node test/test.js
+  # If running in CI, then output .tap files and covert them to JUnit
+  # format for test reporting.
+  local testArgs=""
+  if [[ -n "$CI" ]]; then
+    rm -rf ./test_output
+    mkdir ./test_output
+    testArgs="-o ./test_output"
+  fi
+  if [[ -n "$COVERAGE" ]]; then
+    nyc node test/test.js $testArgs
   else
-    nyc node test/test.js
+    node test/test.js $testArgs
+  fi
+  if [[ -n "$CI" ]]; then
+    ls test_output/*.tap | while read f; do cat $f | ./node_modules/.bin/tap-junit > $f.junit.xml; done
   fi

   npm run test:types
```